### PR TITLE
Migrate `docs.datajoint.org/python`

### PR DESCRIPTION
**Pull request by kabilar**
## Description
Currently, DataJoint Python's [new docs](https://datajoint.com/docs/core/datajoint-python/) contain only a portion of the content of the [legacy docs](https://docs.datajoint.org/python).  This pull request migrates the remaining content.  After this pull request is merged, our team will continue to update the new docs with the latest developments.

## Changes
- [x] Migrate https://docs.datajoint.org/python/ to https://datajoint.com/docs/core/datajoint-python/
  - [x] Convert reStructuredText to Markdown
  - [x] Update navigation
- [x] Update GitHub Actions
  - [x] Add `docs` workflow to manually release the docs
  - [x] Merge `codespell` workflow into `development` workflow
- [x] Update changelog
- [x] Update readme
- [x] Test the deployed site locally
- After this pull request is merged, I will manually make a new docs release

### CODE REVIEW

**Review by dimitri-yatsenko — APPROVED** [review]

File: docs/src/sysadmin/hosting.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+A cloud hosting option will be provided through https://works.datajoint.com.
+DataJoint Works simplifies the setup for labs that wish to host their data pipelines in 
+the cloud and allows sharing pipelines between multiple groups and locations.
+Being an open-source solution, other cloud services such as Amazon RDS can also be used 
+in this role, albeit with less DataJoint-centric customization.
+
+## Self hosting
+
+In the most basic configuration, the relational database software and DataJoint are 
+installed onto a single computer which is used by an individual user.
```
Comment (dimitri-yatsenko) [inline docs/src/sysadmin/hosting.md:32]:
```suggestion
In the most basic configuration, a relational database management system (database server) is 
installed on an individual user's personal computer.
```

File: docs/src/sysadmin/hosting.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+To support a small group of users, a larger computer can be used instead and configured 
+for remote access.
+As the number of users grows, individual workstations can be installed with the 
+DataJoint software and used to connect to a larger and more specialized centrally 
+located database server machine.
```
Comment (dimitri-yatsenko) [inline docs/src/sysadmin/hosting.md:37]:
```suggestion
To support a group of users, a specialized machine can be configured as a dedicated database server.
This server can be accessed by multiple DataJoint clients to query the data and perform computations. 
```

File: docs/src/sysadmin/hosting.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+For even larger groups or multi-site collaborations, multiple database servers may be 
+configured in a replicated fashion to support larger workloads and simultaneous 
+multi-site access.
```
Comment (dimitri-yatsenko) [inline docs/src/sysadmin/hosting.md:41]: For larger groups and multi-site collaborations with heavy workloads, the database server cluster may be configured in the cloud or on premises.

File: docs/src/sysadmin/hosting.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+## General server / hardware support requirements
+
+The following table lists some likely scenarios for DataJoint database server 
+deployments and some reasonable estimates of the required computer hardware.
+The required IT/systems support needed to ensure smooth operations in the absence of 
+local database expertise is also listed.
+
+### IT infrastructures
+
+| Usage Scenario | DataJoint Database Computer | Required IT Support |
```
Comment (dimitri-yatsenko) [inline docs/src/sysadmin/hosting.md:54]: Hm, how was this generated? Sources?

**Review by dimitri-yatsenko — APPROVED** [review]

File: docs/src/sysadmin/hosting.md (hunk 4 above)
Comment (kabilar) [reply docs/src/sysadmin/hosting.md:54]: @dimitri-yatsenko This is just a direct migration from: https://docs.datajoint.org/python/admin/1-hosting.html

Comment (dimitri-yatsenko) [reply docs/src/sysadmin/hosting.md:54]: ok, let's migrate and I will edit with references to sources. Thanks.

File: docs/src/sysadmin/hosting.md (hunk 3 above)
Comment (kabilar) [reply docs/src/sysadmin/hosting.md]: Thanks.  Updated.

**Review by dimitri-yatsenko — APPROVED** [review]

**Comment by kabilar** [comment]
These changes have been deployed to [here](https://datajoint.com/docs/core/datajoint-python/).

**Pull request merged**